tonic/src/codec/encode.rs

```diff
@@ -261,18 +206,20 @@ fn finish_encoding(
     Ok(())
 }
 
-#[derive(Debug)]
-enum Role {
+/// Role for use in encoding a body.
+#[allow(missing_docs)]
+#[derive(Clone, Copy, Debug)]
+pub enum Role {
     Client,
     Server,
 }
```

Comment:
Why? Do you foresee any changes in the near future that will need changes to this type?

Reply:
The reason I was wondering about this is because it doesn't seem to have to be an enum. It doesn't look like we don't expect any changes in the near future, so it shouldn't be a problem.